### PR TITLE
Add edit concert command

### LINKED ISSUE #101: Edit concert command

### PR DESCRIPTION
**Pull request by wes-w-z-h**
Fixes #101

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditConcertCommand.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            }
+        }
+
+        model.setConcert(concertToEdit, editedConcert);
+        model.updateFilteredConcertList(PREDICATE_SHOW_ALL_CONCERTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_CONCERT_SUCCESS, Messages.format(editedConcert)));
+    }
+
+    /**
+     * Creates and returns a {@code Cocnert} with the details of {@code concertToEdit}
```
Comment (karthickkc) [inline src/main/java/seedu/address/logic/commands/EditConcertCommand.java]: same here

File: src/main/java/seedu/address/logic/commands/EditConcertCommand.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+    public static final String MESSAGE_EDIT_CONCERT_SUCCESS = "Edited Concert: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_CONCERT = "This concert already exists in the address book";
+
+    private final Index index;
+    private final EditConcertDescriptor editConcertDescriptor;
+
+    /**
+     * @param index of the conert to edit.
```
Comment (karthickkc) [inline src/main/java/seedu/address/logic/commands/EditConcertCommand.java]: minor typo should be concert

File: src/main/java/seedu/address/logic/commands/EditConcertCommand.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("index", index)
+                .add("editConcertDescriptor", editConcertDescriptor)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the concert with. Each non-empty field value will repalce the
```
Comment (karthickkc) [inline src/main/java/seedu/address/logic/commands/EditConcertCommand.java]: typo should be replace

**Review by karthickkc — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/EditConcertCommand.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME" + "] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS" + "] "
+            + "[" + PREFIX_DATE + "DATE" + "]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Coachella"
+            + PREFIX_ADDRESS + "1 Stadium drive"
+            + PREFIX_DATE + "2024-10-11 2200";
+
```
Comment (karthickkc) [inline src/main/java/seedu/address/logic/commands/EditConcertCommand.java:44]: also for this one there should be space before the prefix_address and prefix_date

**Review by karthickkc — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/EditConcertCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditConcertCommand;
+import seedu.address.logic.commands.EditConcertCommand.EditConcertDescriptor;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments adn creates a new EditConcertCommand object.
```
Comment (bbbbcccd) [inline src/main/java/seedu/address/logic/parser/EditConcertCommandParser.java]: Typo adn -> and

**Review by bbbbcccd — DISMISSED** [review]
lgtm with 1 typo

**Pull request merged**